File: Sources/ATTNPrice.m
//
//  ATTNPrice.m
//  attentive-ios-sdk
//
//  Created by Wyatt Davis on 12/16/22.
//

#import "ATTNPrice.h"

NS_ASSUME_NONNULL_BEGIN

@implementation ATTNPrice

- (instancetype)initWithPrice:(NSDecimalNumber*)price currency:(NSString*)currency {
    if (self = [super init]) {
        self->_price = price;
        self->_currency = currency;
    }
    
    return self;
}

@end

NS_ASSUME_NONNULL_END

File: Sources/Public/Events/ATTNPrice.swift
//
//  ATTNPrice.swift
//  attentive-ios-sdk-framework
//
//  Created by Vladimir - Work on 2024-05-30.
//

import Foundation

@objc(ATTNPrice)
public final class ATTNPrice: NSObject {
  @objc public let price: NSDecimalNumber
  @objc public let currency: String

  @objc(initWithPrice:currency:)
  public init(price: NSDecimalNumber, currency: String) {
    self.price = price
    self.currency = currency
    super.init()
  }

  private override init() {
    fatalError("init() has not been implemented")
  }
}
